modeling/preprocessing.py

```diff
@@ -82,9 +80,30 @@ def convert_to_floats(df, descriptions):
 
     return df_modified
 
+
+
 def handle_outliers(df):
     """
     This function will detect outliers in the data. and replace them with missing values
     """
     # TODO: Implement this function
-    return df
+    df_modified = df.copy()
+    column_names = df.columns.tolist()
+    score_headers = []
+    for column in column_names:
+        if '_score' in column:
+            score_headers.append(column) 
+    for score in score_headers: 
+        df_copy_score = df_modified[score]
+        upperQuartile = df_copy_score.quantile(.75)
+        lowerQuartile = df_copy_score.quantile(.25)
+        iqr = upperQuartile - lowerQuartile
+        for index, row in df_modified.iterrows():
```

Comment:
Ideally, we want to try avoid loops in pandas due to slower speeds where we can. Instead, we'd like to use vectorized operations. Read [here](https://towardsdatascience.com/vectorization-must-know-technique-to-speed-up-operations-100x-faster-50b6e89ddd45): 

You can drop this code into chatgpt or copilot ask it to vectorize this loop, to see how that may be done.